plugin/src/test/java/io/jenkins/plugins/casc/SchemaGenerationTest.java

```diff
@@ -1,60 +1,44 @@
-import org.everit.json.schema.ValidationException;
-import org.everit.json.schema.loader.SchemaLoader;
+import java.util.logging.Logger;
 import org.json.JSONObject;
 import org.json.JSONTokener;
 import org.junit.Rule;
 import org.junit.Test;
 
-import static io.jenkins.plugins.casc.SchemaGeneration.generateSchema;
-import static org.junit.Assert.fail;
+import static io.jenkins.plugins.casc.misc.Util.validateSchema;
+import static org.junit.Assert.assertFalse;
+import static org.junit.Assert.assertTrue;
 
 public class SchemaGenerationTest {
+
+    private static final Logger LOGGER = Logger.getLogger(SchemaGenerationTest.class.getName());
+    private static final String stringToYAMLConversionFailure  = "Error Converting the YAML file to String";
+
     @Rule
     public JenkinsConfiguredWithCodeRule j = new JenkinsConfiguredWithCodeRule();
 
     @Test
-    public void validSchemaShouldSucceed() throws Exception {
-        JSONObject schemaObject = generateSchema();
-        JSONObject jsonSchema = new JSONObject(
-            new JSONTokener(schemaObject.toString()));
-        String yamlStringContents = Util.toStringFromYamlFile(this, "validSchemaConfig.yml");
-        JSONObject jsonSubject = new JSONObject(
-            new JSONTokener(Util.convertToJson(yamlStringContents)));
-        Schema schema = SchemaLoader.load(jsonSchema);
-        try {
-            schema.validate(jsonSubject);
-        } catch (Exception e) {
-            fail(e.getMessage());
-        }
+    public void validSchemaShouldSucceed() {
+        assertTrue(validateSchema(convertYamlFileToJson("validSchemaConfig.yml")));
     }
 
     @Test
-    public void invalidSchemaShouldNotSucceed() throws Exception {
-        JSONObject schemaObject = generateSchema();
-        Gson gson = new GsonBuilder().setPrettyPrinting().create();
-        JsonParser jsonParser = new JsonParser();
-        JsonElement jsonElement = jsonParser.parse(schemaObject.toString());
-        String prettyJsonString = gson.toJson(jsonElement);
-        JSONObject jsonSchema = new JSONObject(
-            new JSONTokener(prettyJsonString));
-        String yamlStringContents = Util.toStringFromYamlFile(this, "invalidSchemaConfig.yml");
-        JSONObject jsonSubject = new JSONObject(
-            new JSONTokener(Util.convertToJson(yamlStringContents)));
-        Schema schema = SchemaLoader.load(jsonSchema);
+    public void invalidSchemaShouldNotSucceed() {
+        assertFalse(validateSchema(convertYamlFileToJson("invalidSchemaConfig.yml")));
+
+    }
+
+    private JSONObject convertYamlFileToJson(String yamlFile)  {
+        String yamlStringContents = null;
         try {
-            schema.validate(jsonSubject);
-            fail();
-        } catch (ValidationException ve) {
-            ve.printStackTrace();
+            yamlStringContents = Util.toStringFromYamlFile(this, yamlFile);
+        } catch (Exception e) {
+            LOGGER.warning(stringToYAMLConversionFailure);
```

Comment:
shouldn't this just throw an exception? or is there a reason you're logging and continuing?

Reply:
Should I return ? after logging ? or do not log at all ?

Reply:
I would throw some form of runtime  exception

Reply:
Yeah I will throw an exception